mne/epochs.py

```diff
@@ -803,7 +809,10 @@ def iter_evoked(self):
                 break
             data, event_id = out
             tmin = self.times[0]
-            info = deepcopy(self.info)
+            info = self.info
+            if copy:
+                info = deepcopy(self.info)
+                data = data.copy()
```

Comment:
@larsoner are you sure there was not a copy already before? This now may change old behavior so could be an API change

Reply:
Arguably then it’d be a bugfix

Reply:
The data was not copied before, it was a view. So on `master` data is not copied, but `info` is copied. This is inconsistent and should be fixed. The current version of this PR fixes it so either info and data are copied, or neither are.

My vote would be to have `copy=False` as the default, since that was closest to what was being done before in terms of the data, which is probably what most users think about. But I doubt the number of people doing in-place modifications with an `iter_evoked` is quite small.